packages/serverpod/lib/src/server/tasks/task_manager.dart

```diff
@@ -0,0 +1,73 @@
+import 'dart:async';
+
+import 'package:meta/meta.dart';
+
+import 'task.dart';
+
+/// Manages tasks that need to be executed.
+///
+/// The `TaskManager` allows registering tasks that can be executed
+/// concurrently. Each task is identified by a unique ID and contains
+/// a callback function that will be executed when the task is run.
+abstract interface class TaskManager {
+  /// Adds a task to be executed.
+  ///
+  /// The task is stored in the task map with its ID as the key.
+  /// The [task] contains an ID for identification and a callback function
+  /// that will be executed when the task is run.
+  ///
+  /// Throws [StateError] if a task with the same ID already exists.
```

Comment:
Was at first surprised that an `add*` would throw on duplicate (imagining a list underneath as my mental model for the task list).

But good with the exception, then at least no developer would overwrite a task by accident, and they would be steered toward `removeTask` + `addTask` in case they wanted to update one.